src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [fromParts, fromSeparator] = [
+    fromFormat.slice(0, -1),
+    fromFormat.slice(-1)[0],
+  ];
+  const [toParts, toSeparator] = [toFormat.slice(0, -1), toFormat.slice(-1)[0]];
```

Comment:
Similar to the `fromFormat`, the assumption that the last element of the `toFormat` array is the separator might not always be correct. Consider validating the format or explicitly defining the separator.